src/pages/EditMonitor.vue

```diff
@@ -1157,6 +1157,23 @@ export default {
     },
 
     computed: {
+        defaultFriendlyName() {
+            if (this.monitor.hostname) {
+                return this.monitor.hostname;
+            }
+            if (this.monitor.url) {
+                try {
+                    const url = new URL(this.monitor.url);
+                    return url.hostname;
+                } catch (e) {
+                    // Handle invalid URL, maybe return a generic placeholder or empty string
+                    return this.$t("My Monitor");
+                }
+            }
+            // Default placeholder if neither hostname nor URL is available
+            return this.$t("My Monitor");
```

Comment:
Please add all translation strings to `en.json`. If not included there, our translators cannot tanslate this.

What do you think about

`Monitor #42` instead of `My Monitor`?

```suggestion
            return this.$t("defaultFriendlyName", [this.monitor.id]);
```

and in en.json:
```json
"defaultFriendlyName": "Monitor #{0}"
```

Reply:
`Monitor #42` is indeed a better choice, but the `monitor.id` can only be determined **after submitting to the backend**. 

An alternative would be using max(id)+1 from the frontend's available monitor items, but this isn't reliable since the max ID might have been deleted.

Possible approaches I can think of:
1. A fixed default value (`New Monitor` or similar), or
2. A timestamp-based name like `Monitor MM-DD HH:MM`

Interested to hear your feedback. Thanks.

Reply:
Hmmm.. I don't like it.
I guess `New Monitor` is what I can live with most..